23_tiles_msleckman/3_visualize/src/raster_plotting_w_ggplot.R

```diff
@@ -1,26 +1,124 @@
-raster_ploting_w_ggplot <- function(raster_in, reach_shp, out_folder = "3_visualize/out/"){
+raster_ploting_w_ggplot <- function(raster_in, reach_shp, counts, legend_df, title, font_fam = "Source Sans Pro", out_folder = "3_visualize/out/"){
   
-  #raster_in <- p2_reclassified_raster_list[[1]]
-  raster_name <- names(raster_in)
+  font_legend <- 'Source Sans Pro'
   
-  #
-  raster_df <- as.data.frame(raster_in, xy = TRUE) %>%
-    na.omit() %>%
-    rename(value = {{raster_name}})
+  nlcd_map <- ggplot()+
+    geom_raster(data = raster_in, 
+                aes(x=x, y=y, fill = factor(name))) +
+    geom_sf(data = reach_shp %>%
+              filter(streamorde > 2), 
+            color = "white", 
+            aes(size = streamorde)) +
+    theme_void() +
+    theme(
+      #legend.position = "none",
+      text = element_text(family = font_legend),
+      panel.background = element_rect(fill = "white", color = NA),
+      plot.background = element_rect(fill = "white", color = NA)
+    )+
+    scale_size(
+      breaks = c(5, 6, 7),
+      range = c(0.2, 0.8),
+      guide = "none"
+    ) +
+    scale_fill_manual(
+      values = legend_df$color,
+      labels = legend_df$Reclassify_description,
+      "Land cover"
+    ) +
+    coord_sf()
   
-  raster_name <- names(raster_in)
-  frame_out <- paste0(out_folder, "ggplot_", raster_name,'.png')
-  png(frame_out)
   
-  raster_plot <- ggplot()+
-    geom_raster(data = raster_df, aes(x = x, y = y, fill = value))+
-    geom_sf(data = reach_shp, color = 'blue')
+  # area through time
+  nlcd_area <- counts %>% 
+    # find % of total area in each category over time
+    left_join(counts %>% 
+                group_by(rast)%>%
+                summarize(total_cells = sum(count))) %>%
+    mutate(year = as.numeric(stringr::word(rast, 4, 4, sep ='_')),
```

Comment:
> My pipeline is only producing frames up until 2000 and I'm not sure why, I may have incompletely pulled down your fork. I'm also using a funny syntax in some of the dynamic branching that works, but I don't know why

It's not catching the years post 2000 because the filenames are different and have different 3 words instead of 4. suggest grabbing the last 4 characters. 

```suggestion
    mutate(year = as.numeric(stringr::str_sub(rast,-4,-1)),
```